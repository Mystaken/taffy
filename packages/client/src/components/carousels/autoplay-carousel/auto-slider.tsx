import React, { FunctionComponent, useState } from 'react';
import SwipeableViews from 'react-swipeable-views';
import { autoPlay, WithAutoPlayProps } from 'react-swipeable-views-utils';
import { MobileStepper } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';

import { arrayify } from '../../../utils/common';

// TODO (Low Priority): May need to redo the pagination buttons
const AutoplaySwipeableViews = autoPlay(SwipeableViews);

export interface AutoSliderProps extends Partial<WithAutoPlayProps> {
  /**
   * If set to `true`, pagination dots will be added.
   */
  pagination?: boolean;
  /**
   * If set to `true`, panels will be swipable.
   */
  swipeable?: boolean;
}

const autoSliderStyles = makeStyles(_ => ({
  wrapper: {
    position: 'relative'
  },
  pagination: {
    position: 'absolute',
    bottom: 8,
    right: 8,
    display: 'flex',
    flexDirection: 'row',
    backgroundColor: 'inherit'
  }
}));

export const AutoSlider: FunctionComponent<AutoSliderProps> = ({
  pagination = true,
  swipeable = true,
  ...props
}) => {
  const [index, setIndex] = useState(0);
  const classes = autoSliderStyles();
  const children = arrayify(props.children);
  return (
    <div className={classes.wrapper}>
      <AutoplaySwipeableViews
        index={index}
        onChangeIndex={setIndex}
        enableMouseEvents
        {...props}
      />
      {pagination && (
        <MobileStepper
          className={classes.pagination}
          variant="dots"
          steps={children.length}
          backButton={<></>}
          nextButton={<></>}
          position="static"
          activeStep={index}
        />
      )}
    </div>
  );
};
