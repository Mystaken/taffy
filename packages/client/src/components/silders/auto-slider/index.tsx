import React, { FunctionComponent, useState } from 'react';
import SwipeableViews from 'react-swipeable-views';
import { autoPlay, AutoPlayProps } from 'react-swipeable-views-utils';
import { MobileStepper } from '@material-ui/core';
import { arrayify } from '../../../utils/common';
import { makeStyles } from '@material-ui/styles';

const AutoplaySwipeableViews = autoPlay(SwipeableViews);

export interface AutoSliderProps extends AutoPlayProps {
  /**
   * If set to `true`, pagination dots will be added.
   */
  pagination?: boolean;
  /**
   * If set to `true`, panels will be swipable.
   */
  swipeable?: boolean;
}

const autoSliderStyles = makeStyles(_ => ({
  wrapper: {
    position: 'relative'
  },
  pagination: {
    position: 'absolute',
    bottom: 8,
    right: 8,
    display: 'flex',
    flexDirection: 'row',
    backgroundColor: 'inherit'
  }
}));

export const AutoSlider: FunctionComponent<AutoSliderProps> = ({
  pagination = true,
  swipeable = true,
  ...props
}) => {
  const [index, setIndex] = useState(0);
  const classes = autoSliderStyles();
  const children = arrayify(props.children);
  return (
    <div className={classes.wrapper}>
      <AutoplaySwipeableViews
        index={index}
        onChangeIndex={setIndex}
        enableMouseEvents
        {...props}></AutoplaySwipeableViews>
      {pagination && (
        <MobileStepper
          className={classes.pagination}
          variant="dots"
          steps={children.length}
          backButton={<></>}
          nextButton={<></>}
          position="static"
          activeStep={index}></MobileStepper>
      )}
    </div>
  );
};
