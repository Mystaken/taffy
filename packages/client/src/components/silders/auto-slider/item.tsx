import React, { FunctionComponent, MouseEvent } from 'react';
import { CardMedia } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';

export interface AutoSliderItemProps {
  /** The image url to display. */
  image: string;
  onClick?: (event: MouseEvent<HTMLElement>) => void;
}

const useStyles = makeStyles(_ => ({
  card: {
    width: '100%',
    position: 'relative',
    paddingTop: '56.25%',
    cursor: 'pointer'
  }
}));

export const AutoSliderItem: FunctionComponent<AutoSliderItemProps> = props => {
  const classes = useStyles();
  return <CardMedia className={classes.card} {...props} />;
};
