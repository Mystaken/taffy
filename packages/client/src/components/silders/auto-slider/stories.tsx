import { storiesOf } from '@storybook/react';
import { AutoSlider } from '.';

const stories = storiesOf('Sliders', module);
stories.addParameters({ info: { inline: true } });

const styles = {
  slide: {
    padding: 15,
    minHeight: 100,
    color: '#fff'
  },
  slide1: {
    backgroundColor: '#FEA900'
  },
  slide2: {
    backgroundColor: '#B3DC4A'
  },
  slide3: {
    backgroundColor: '#6AC0FF'
  }
};

const slideRenderer = (index: number) => {
  let style;

  switch (Math.abs(index) % 3) {
    case 0:
      style = styles.slide1;
      break;

    case 1:
      style = styles.slide2;
      break;

    case 2:
      style = styles.slide3;
      break;

    default:
      break;
  }

  return (
    <div style={{ ...styles.slide, ...style }} key={index}>
      {`Slide #${index + 1}`}
    </div>
  );
};

stories.add('Auto', () => (
  <AutoSlider autoplay={true}>
    {[0, 1, 2, 3, 4, 5, 6, 7, 8, 9].map(slideRenderer)}
  </AutoSlider>
));
