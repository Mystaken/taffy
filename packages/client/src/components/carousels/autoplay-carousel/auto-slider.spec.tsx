import React from 'react';
import { AutoSlider } from './auto-slider';
import { shallow } from 'enzyme';

describe('<AutoSlider />', () => {
  it('has rendered correctly', () => {
    const tree = shallow(<AutoSlider />);
    expect(tree).toMatchSnapshot();
  });
  it('has rendered correctly with props', () => {
    const tree = shallow(<AutoSlider pagination swipeable />);
    expect(tree).toMatchSnapshot();
  });
});
