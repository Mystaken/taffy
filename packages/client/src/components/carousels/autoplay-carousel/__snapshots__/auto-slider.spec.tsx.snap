// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutoSlider /> has rendered correctly 1`] = `ShallowWrapper {}`;

exports[`<AutoSlider /> has rendered correctly with props 1`] = `ShallowWrapper {}`;
